src/inject/dynamic-theme/style-manager.ts

```diff
@@ -318,44 +318,67 @@ export function manageStyle(element: StyleElement, {update, loadingStart, loadin
         return cssRules;
     }
 
-    let rulesChangeKey: number = null;
-    let rulesCheckFrameId: number = null;
+    function subscribeToSheetChanges() {
 
-    function updateRulesChangeKey() {
-        const rules = safeGetSheetRules();
-        if (rules) {
-            rulesChangeKey = rules.length;
-        }
-    }
+        element.addEventListener('updateSheet', () => {
+            update();
+        });
+
+        let script = document.head.querySelector('.darkreader-proxy') as HTMLScriptElement;
+        if (!script) {
+            script = document.createElement('script');
+            script.classList.add('darkreader');
+            script.classList.add('darkreader-proxy');
+            script.textContent = `
+                // First get the originial 'native code' of the function.
+                const addRuleFunction = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, 'addRule');
+                const insertRuleFunction = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, 'insertRule');
+                const deleteRuleFunction = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, 'deleteRule');
+                const removeRuleFunction = Object.getOwnPropertyDescriptor(CSSStyleSheet.prototype, 'removeRule');
+
+                // Create a new Event so it can be dispatched.
+                const event = new Event('updateSheet');
+
+                // Create the proxy function that should be identicial to the native one's.
+
+                function proxyAddRule(selector, style, index) {
+                    addRuleFunction.value.call(this, selector, style, index);
+                    this.ownerNode.dispatchEvent(event);
+                    // Check why it returns -1 https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/addRule#Return_value.
+                    return -1;
+                }
+                function proxyInsertRule(rule, index) {
+                    // Cache the return value of insertRule as insertRule has the chance to not return the originial index.
```

Comment:
https://github.com/chromium/chromium/blob/master/third_party/blink/renderer/core/css/css_style_sheet.cc#L322-L367 can return `0` due to errors.
https://searchfox.org/mozilla-central/source/layout/style/StyleSheet.cpp#588-597 can return `0` due to errors.